component/service/src/main/java/org/exoplatform/social/rest/impl/site/SiteRest.java

```diff
@@ -135,13 +155,81 @@ public Response getSites(@Context
+      @ApiResponse(responseCode = "404", description = "Resource not found") })
+  public Response getSiteBanner(@Context
+  Request request,
+                                @Parameter(description = "site name", required = true)
+                                @PathParam("siteName")
+                                String siteName,
+                                @Parameter(description = "A mandatory valid token that is used to authorize anonymous request", required = true)
+                                @QueryParam("r")
+                                String token) {
+    PortalConfig portalConfig = layoutService.getPortalConfig(siteName);
+    if (portalConfig == null) {
+      return Response.status(Response.Status.NOT_FOUND).entity("The site doesn't exit").build();
+    }
+    boolean isDefault = portalConfig.getBannerFileId() == 0;
+    try {
+      if (!isDefault && RestUtils.isAnonymous()
+          && !RestUtils.isAttachmentTokenValid(token, siteName, RestUtils.ATTACHMENT_BANNER_TYPE)) {
+        LOG.warn("An anonymous user attempts to access banner of site {} without a valid access token", siteName);
+        return Response.status(Response.Status.FORBIDDEN).build();
+      }
+      EntityTag eTag = isDefault ? new EntityTag(siteName + "-" + RestUtils.ATTACHMENT_BANNER_TYPE)
+                                 : new EntityTag(String.valueOf(portalConfig.getBannerFileId()));
+      Response.ResponseBuilder builder = request.evaluatePreconditions(eTag);
+      if (builder == null) {
+        InputStream stream = isDefault ? getDefaultBannerInputStream(portalConfig.getName())
+                                       : layoutService.getSiteBannerStream(new SiteKey(portalConfig.getType(),
+                                                                                       portalConfig.getName()));
+        builder = Response.ok(stream, "image/png");
+        builder.tag(eTag);
+      }
+      builder.cacheControl(CACHECONTROL);
+      builder.expires(new Date(System.currentTimeMillis() + CACHE_IN_MILLI_SECONDS));
+
+      return builder.build();
+    } catch (ObjectNotFoundException | IOException e) {
+      return Response.status(Response.Status.NOT_FOUND).build();
+    }
+  }
+
+  private InputStream getDefaultBannerInputStream(String siteName) throws IOException {
+    byte[] defaultSiteBanner = null;
+    InputStream is =  portalContainer.getPortalContext()
+                   .getResourceAsStream(System.getProperty("sites." + siteName + ".defaultBannerPath",
+                                                           "/images/sites/" + siteName.toLowerCase() + ".png"));
+    if (is == null) {
+      is = portalContainer.getPortalContext().getResourceAsStream(SITE_DEFAULT_BANNER_URL);
+      defaultSiteBanner = is != null ? IOUtil.getStreamContentAsBytes(is) : new byte[] {};
+    } else {
+      defaultSiteBanner = IOUtil.getStreamContentAsBytes(is);
+    }
+    return new ByteArrayInputStream(defaultSiteBanner);
```

Comment:
why not returning directly is ?

Reply:
can you consider this https://github.com/Meeds-io/social/pull/2851/files#r1326065689 ?